### ISSUE TITLE
Gemini: как снять скриншот после timeout даже если условие не выполнено?

### ISSUE THREAD

**Issue by varya** [bem/bem-forum-content-ru#512]
Делаю скриншоты нескольких страниц (условия одинаковые). Скриншоты делаются везде, кроме одной страницы.
Чтобы понять почему, я бы хотела видеть картинку: что видно после timeout. Поскольку дебажить нормально пока нельзя, я хоть какую-то информацию из этого получу. Это как-то возможно сделать?
/cc @arikon @sevinf

**Comment by scf2k** [comment]
Какой timeout?

**Comment by arikon** [comment]
@varya Это js condition timeout?

> Чтобы понять почему, я бы хотела видеть картинку: что видно после timeout. Поскольку дебажить нормально пока нельзя, я хоть какую-то информацию из этого получу.

@SevInf Тоже вчера про подобное думал: может, будет в случае ошибки снимать всю страницу (если это возможно), и выводить ссылку на картинку в консоль?

> Это как-то возможно сделать?

@varya Сейчас вроде бы нет.

**Comment by varya** [comment]
> @varya Это js condition timeout?
@arikon `waitForJSCondition` и `waitForElementToShow`. И может быть ещё какие-то, если есть, но я пользовалась только этими двумя.

**Comment by varya** [comment]
@arikon

Кстати, с `waitForJSCondition` это сейчас возможно сделать вот так:

```
var gemini = require('gemini');

var examples = [
  '1.1',
  '1.2-1',
  '2.1'
];

examples.forEach(function(example) {

  gemini.suite(example, function(suite) {
      suite.setUrl('/styleguide/#/section/' + example + '/fullscreen')
          .setCaptureElements('body')
          .capture('plain', function(actions, find){
              actions.executeJS(function(window) {

                  window.addEventListener('my event', function() {
                      window.doTake = true;
                  });
                  window.setTimeout(function() {
                    window.doTake = true;
                  }, 10000);

              });

              actions.waitForJSCondition(function(window) {
                  return window.doTake;
              }, 10500);
          });
  });

});

```

Но с `waitForElementToShow` естественно нельзя :-)

**Comment by arikon** [comment]
@SevInf Здесь нужно твоё мнение

**Comment by SevInf** [comment]
@arikon да, есть в планах при ошибках показывать полный скриншот. Хотелось бы в финальной 0.13 сделать.

@varya пока это еще не сделано, можно попробовать вручную заменить waitForElementToShow на wait(timeout) и посмотреть что там происходит.
